tools/webdriver/webdriver/bidi/modules/network.py

```diff
@@ -57,11 +59,97 @@ def _add_intercept(self, result: Mapping[str, Any]) -> Any:
         assert result["intercept"] is not None
         return result["intercept"]
 
+    @command
+    def continue_with_auth(
+            self,
+            request: str,
+            action: str,
+            credentials: Optional[Dict[str, str]] = None) -> Mapping[str, Any]:
+        params: MutableMapping[str, Any] = {
+            "request": request,
+            "action": action,
+        }
+
+        if action == "provideCredentials" and credentials is not None:
+            if "type" in credentials:
+                params["type"] = credentials["type"]
+            if "username" in credentials:
+                params["username"] = credentials["username"]
+            if "password" in credentials:
+                params["password"] = credentials["password"]
+
+        return params
+
+    @command
+    def continue_request(self,
+                         request: str,
+                         method: Optional[str] = None,
+                         url: Optional[str] = None) -> Mapping[str, Any]:
+        params: MutableMapping[str, Any] = {
+            "request": request,
+        }
+
+        if method is not None:
+            params["method"] = method
+
+        if url is not None:
+            params["url"] = url
+
+        # TODO: Add body.
+        # TODO: Add cookies.
+        # TODO: Add headers.
```

Comment:
Is there a reason why you didn't add those yet?

Reply:
It's because we need to add their type definitions, which are not trivial. Just didn't have time to get to them yet. 

Also, these are not yet implemented in our chromium mapper implementation, and I wanted to be confident that the tests I am adding are passing, so until we have a way to test these fields it's better to refrain from adding them. Are they already added to Gecko / Firefox?

Reply:
No, not at this point and when Julian is back we will most likely continue to have the authentication feature done first. So that's fine then.

Reply:
Ack